src/test/java/com/google/firebase/auth/FirebaseAuthIT.java

```diff
@@ -705,6 +706,60 @@ public void testCustomTokenWithIAM() throws Exception {
     }
   }
 
+  @Test
+  public void testTenantAwareCustomToken() throws Exception {
+    // Create tenant to use.
+    TenantManager tenantManager = auth.getTenantManager();
+    Tenant.CreateRequest tenantCreateRequest =
+        new Tenant.CreateRequest().setDisplayName("DisplayName");
+    String tenantId = tenantManager.createTenant(tenantCreateRequest).getTenantId();
+
+    try {
+      // Create and decode a token with a tenant-aware client.
+      TenantAwareFirebaseAuth tenantAwareAuth = auth.getTenantManager().getAuthForTenant(tenantId);
+      String customToken = tenantAwareAuth.createCustomTokenAsync("user1").get();
+      String idToken = signInWithCustomToken(customToken, tenantId);
+      FirebaseToken decoded = tenantAwareAuth.verifyIdTokenAsync(idToken).get();
+      assertEquals("user1", decoded.getUid());
+      assertEquals(tenantId, decoded.getTenantId());
+    } finally {
+      // Delete tenant.
+      tenantManager.deleteTenantAsync(tenantId).get();
+    }
+  }
+
+  @Test
+  public void testVerifyTokenWithWrongTenantAwareClient() throws Exception {
+    // Create tenants to use.
+    TenantManager tenantManager = auth.getTenantManager();
+    Tenant.CreateRequest tenantCreateRequest1 =
+        new Tenant.CreateRequest().setDisplayName("DisplayName1");
+    String tenantId1 = tenantManager.createTenant(tenantCreateRequest1).getTenantId();
+    Tenant.CreateRequest tenantCreateRequest2 =
```

Comment:
You can only create one tenant for this test. For the 2nd tenant use a fake tenant ID string. And to ensure proper clean up you will have to do something like this:

```
// create test tenants
try {
   // create token
   try {
     // verify token
   }. catch {
     // assert
   }
} finally {
  // clean up tenants
}
```

You can move the inner try-catch into a new helper method if this looks too ugly.

Reply:
Done.